src/main/kotlin/com/sourcegraph/cody/edit/FixupService.kt

```diff
@@ -0,0 +1,166 @@
+            } else {
+              session.taskDeleted()
+            }
+          }
+
+          agent.client.setOnWorkspaceEdit { params ->
+            for (op in params.operations) {
+              when (op.type) {
+                "create-file" -> {
+                  logger.warn("Workspace edit operation created a file: ${op.uri}")
+                }
+                "rename-file" -> {
+                  logger.warn(
+                      "Workspace edit operation renamed a file: ${op.oldUri} -> ${op.newUri}")
+                }
+                "delete-file" -> {
+                  logger.warn("Workspace edit operation deleted a file: ${op.uri}")
+                }
+                "edit-file" -> {
+                  if (op.edits == null) {
+                    logger.warn("Workspace edit operation has no edits")
+                  } else {
+                    // If there is a pending session, assume that it is the one that caused the
+                    // edit.
+                    val session: FixupSession? =
+                        if (pendingSessions.isNotEmpty()) {
+                          pendingSessions.first()
+                        } else {
+                          // TODO: This is what I'd like to be able to do, but it requires a
+                          // protocol change:
+                          // session = activeSessions[op.id]
+                          activeSessions.values.firstOrNull()
+                        }
+                    if (session == null) {
+                      logger.warn("No sessions found for performing inline edits")
+                    } else {
+                      session.performInlineEdits(op.edits)
+                    }
+                  }
+                }
+                else ->
+                    logger.warn(
+                        "DocumentCommand session received unknown workspace edit operation: ${op.type}")
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  /** Entry point for the inline edit command, called by the action handler. */
+  fun startCodeEdit(editor: Editor) {
+    if (isEligibleForInlineEdit(editor)) {
+      EditCommandPrompt(this, editor, "Edit Code with Cody").displayPromptUI()
+    }
+  }
+
+  /** Entry point for the document code command, called by the action handler. */
+  fun startDocumentCode(editor: Editor) {
+    if (isEligibleForInlineEdit(editor)) {
+      DocumentCodeSession(this, editor)
+    }
+  }
+
+  fun isEligibleForInlineEdit(editor: Editor): Boolean {
+    if (!isCodyEnabled()) {
+      logger.warn("Edit code invoked when Cody not enabled")
+      return false
+    }
+    if (!CodyEditorUtil.isEditorValidForAutocomplete(editor)) {
+      logger.warn("Inline edit invoked when editing not available")
+      return false
+    }
+    return true
+  }
+
+  // TODO: get model list from protocol
+  fun getModels(): List<String> = listOf("GPT-4", "GPT-3.5")
```

Comment:
FYI we are already doing it for LLM selector so maybe you can reuse that code.

Reply:
Yikes, forgot that's not wired up yet -- I've added a [GH issue](https://github.com/sourcegraph/jetbrains/issues/1137) to track it. Thanks.